src/Forms/SearchableDropdownTrait.php

```diff
@@ -586,4 +586,13 @@ private function updateOptionsForSchema(
         }
         return $options;
     }
+
+    public function performReadonlyTransformation()
+    {
+        $field = FormField::castedCopy(LazyLoadedLookupField::class);
```

Comment:
```suggestion
        $field = parent::castedCopy(SearchableLookupField::class);
```

Reply:
I don't think this will work. `parent::castedCopy` will go up the chain from `SearchableDropdownField` to `DropdownField` to `SingleSelectField::castedCopy` which in turn calls `parent::castedCopy` again, ending up in `SelectField::castedCopy` and that calls `$this->getSource()` which brings us back to the original problem.

Thats why I went with its ancestor, FormField, directly.

Reply:
The problem is using FormField, is that while this works, it's giving off a warning about statically calling a non-static method

![image](https://github.com/silverstripe/silverstripe-framework/assets/4809037/0011cf30-e6b6-480f-96fb-6dc6bfe35d9b)

Try changing it to this

```php
        // This is calling a non-static method statically
        // using call_user_func() here to prevent an IDE warning
        // The reason we're calling FormField::castedCopy directly is to prevent an ancestor
        // call to $this->getSource() which will load the entire DataList into memory which
        // causes issues with very large datasets and isn't needed when the field is read-only
        $field = call_user_func('SilverStripe\\Forms\\FormField::castedCopy', LazyLoadedLookupField::class);
```